### PR TITLE
AVM-Review-PR

### PR DESCRIPTION
**Pull request by jchancellor-ms**
## Description

PR for AVM official review

## Type of Change

- [ ] Non-module change (e.g. CI/CD, documentation, etc.)
- [x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates.
  - [x] Breaking changes.
  - [x] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -313,7 +315,7 @@ variable "peerings" {
     ipv4_enabled                  = optional(bool, true)
     shared_key                    = optional(string, null)
     peer_asn                      = optional(number, null)
-    route_filter_id               = optional(string, null)
+    route_filter_resource_id      = optional(string, null)
```
Comment (khushal08) [inline variables.tf:318]: @jchancellor-ms Can you please update main.tf as well?

File: variables.tf [hunk 2]
```diff
@@ -335,62 +337,76 @@ variable "peerings" {
...
+(Optional) A map of association objects to create peerings between the created circuit and the designated gateways. 
+
+- `peering_type` - (Required) The type of peering. Possible values are `AzurePrivatePeering`, `AzurePublicPeering`, and `MicrosoftPeering`.
+- `vlan_id` - (Required) The VLAN ID for the peering.
+- `primary_peer_address_prefix` - (Optional) The primary peer address prefix.
+- `secondary_peer_address_prefix` - (Optional) The secondary peer address prefix.
+- `ipv4_enabled` - (Optional) Is IPv4 enabled for this peering. Defaults to `true`.
+- `shared_key` - (Optional) The shared key for the peering.
+- `peer_asn` - (Optional) The peer ASN.
+- `route_filter_id` - (Optional) The ID of the route filter to associate with the peering.
```
Comment (khushal08) [inline variables.tf:349]: @jchancellor-ms Please update this to route_filter_resource_id as well.

**Review by khushal08 — DISMISSED** [review]
Route filter ID variable changes in main.tf. 
Other than that, looks good.

File: variables.tf (hunk 1 above)
Comment (adammontlake) [reply variables.tf:318]: Updated

File: variables.tf (hunk 2 above)
Comment (adammontlake) [reply variables.tf]: Updated

**Review by jchancellor-ms — COMMENTED** [review]
Looks good.  Thanks for making these changes.  I'll update and close the issue and you can publish.

**Review by adammontlake — APPROVED** [review]

**Pull request closed without merging**